File: .guides/tests/ch-4.js

keyPressedEvent  = null;
window.allowDown = false;

$.getScript(window.location.origin + '/public/js/' + window.testEnv.cmd + '.js?_=' + Date.now())
.done(function (script, status) {
  
  createRandomMaze();
    
  var _down = false;
  
  if(typeof keyPressedEvent  == 'function') {

    player.moveDown = function () {
      _down = true;
    }

    keyPressedEvent('DOWN');
        
    if(_down) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }
    
    _down = false;
    allowDown = false;

    keyPressedEvent('A');
    keyPressedEvent('DOWN');

    if(!allowDown || !_down) {
      return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
    }   
  }
  else {
    return codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.FAILURE, 'Not quite right, try again!');
  }
  
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.SUCCESS, 'Well done!');    

})
.fail(function (jqxhr, settings, exception) {
  console.log(exception);
  codio.setButtonValue(window.testEnv.id, codio.BUTTON_STATE.INVALID, exception.message); 
});

File: .guides/tests/ch-4.py

allowed = False
down = False

def setDownAllowed (val):
  global allowed
  allowed = val

def getDownAllowed():
  global allowed
  return allowed

def showMessage(val):
  pass

def moveDown():
  global down
  down = True
  
try:
  execfile('/home/codio/workspace/public/py/ch-4.py')
  
  keyPressedEvent('A')
  keyPressedEvent('DOWN')

  if not down:
    raise ValueError("Error: incorrect behaviour for when down should be allowed")

  down = False
  allowed = False
  keyPressedEvent('DOWN')

  if down:
    raise ValueError("Error: incorrect behaviour for when down should not be allowed")
  
  print 'well done'
  exit(0)
  
except (IOError, SyntaxError, ValueError) as e:
  print str(e)
  exit(1)
  
print 'Not quite right, try again!'
exit(1)
